RepositoryCompiler/CodeModel/CodeParsers/CSharp/CSharpCodeParser.cs

```diff
@@ -199,6 +204,78 @@ private CaDETClass LinkOuterClass(List<CaDETClass> classes, string containerName
         {
             return classes.FirstOrDefault(c => c.FullName.Equals(containerName));
         }
+        
+        private List<CaDETClass> LinkFieldTypes(List<CaDETClass> classes, List<CaDETField> fields)
+        {
+            List<CaDETClass> fieldTypes = new List<CaDETClass>();
+            foreach (var f in fields)
```

Comment:
Can this loop be replaced with fieldTypes.AddRange(FindTypes(classes, fields.Select(f => f.Type)))?

In light of the above comment I see that this approach is not suitable. We need to change each field to set a new Type for it.